src/measure_theory/constructions/borel_space.lean

```diff
@@ -170,6 +170,60 @@ the `σ`-algebra of measurable sets is exactly the `σ`-algebra generated by ope
 class borel_space (α : Type*) [topological_space α] [measurable_space α] : Prop :=
 (measurable_eq : ‹measurable_space α› = borel α)
 
+namespace tactic
+
+/-- Add instances `borel α : measurable_space α` and `⟨rfl⟩ : borel_space α`. -/
+meta def add_borel_instance (α : expr) : tactic unit :=
+do
+  n1 ← get_unused_name "_inst",
+  to_expr ``(borel %%α) >>= pose n1,
+  reset_instance_cache,
+  n2 ← get_unused_name "_inst",
+  t ← to_expr ``(borel_space %%α),
+  v ← to_expr ``(borel_space.mk rfl),
+  note n2 t v,
+  reset_instance_cache
+
+/-- Given a type `α`, an assumption `i : measurable_space α`, and an instance `[borel_space α]`,
+replace `i` with `borel α`. -/
+meta def borel_to_refl (α i : expr) : tactic unit :=
+do
+  n ← get_unused_name "h",
+  to_expr ``(%%i = borel %%α) >>= assert n,
+  applyc `borel_space.measurable_eq,
+  unfreezing (to_expr ``(%%i) >>= tactic.subst),
+  n1 ← get_unused_name "_inst",
+  to_expr ``(borel %%α) >>= pose n1,
+  reset_instance_cache
+
+/-- Given a type `α`, if there is an assumption `[i : measurable_space α]`, then try to prove
+`[borel_space α]` and replace `i` with `borel α`. Otherwise, add instances
+`borel α : measurable_space α` and `⟨rfl⟩ : borel_space α`. -/
+meta def borelize (α : expr) : tactic unit :=
+do
+  i ← optional (to_expr ``(measurable_space %%α) >>= find_assumption),
+  i.elim (add_borel_instance α) (borel_to_refl α)
+
+namespace interactive
+
+setup_tactic_parser
+
+/-- The behaviour of `borelize α` depends on the existing assumptions on `α`.
+
+- if `α` is a topological space with instances `[measurable_space α] [borel_space α]`, then
+  `borelize α` replaces the former instance by `borel α`;
+- otherwise, `borelize α` adds instances `borel α : measurable_space α` and `⟨rfl⟩ : borel_space α`.
+
+Finally, `borelize [α, β, γ]` runs `borelize α, borelize β, borelize γ`.
+-/
+meta def borelize (ts : parse (list_of texpr <|> ((λ e, list.cons e list.nil) <$> texpr))) :
```

Comment:
Probably better to use https://leanprover-community.github.io/mathlib_docs/init/meta/interactive_base.html#interactive.types.pexpr_list_or_texpr for the parser.

Reply:
I tried
```
meta def borelize (ts : pexpr_list_or_texpr) : tactic unit :=
```
and get an error
```
type expected at
  pexpr_list_or_texpr
term has type
  parser_state → interaction_monad.result parser_state (list pexpr)
```
What do I do wrong?

Reply:
It should be `(ts : parse pexpr_list_or_texpr)`. This should be equivalent to what you have now.